### ISSUE TITLE
num of gpus

### ISSUE THREAD

**Issue by sxwenny** [yxgeee/SpCL#1]
Hi, I tried to run this program with two GPUs, and it turned out to be very different from the result of 4 GPUs, and the loss of using 2 GPUs drops quickly. Is this program related to the number of GPUs？

**Comment by yxgeee** [comment]
Hi, re-ID experiments are somewhat sensitive to the number of batch size and GPUs. We use 4 GPUs empirically, since such a setting generally achieves optimal performance. I did not test it on 2 GPUs, how does it perform? I will further look into how to decrease its sensibility.

**Comment by sxwenny** [comment]
I tested on market1501 to dukemtmc, map decreased by about 5%.

**Comment by sxwenny** [comment]
Hi, when I train the model, sometimes there are two problems:
1. It got stuck, not running anymore. 
2. Warning "OMP: Warning #190: Forking a process while a parallel region is active is potentially unsafe."
My experimental environment: cuda 10.0, python 3.7.3, torch 1.2.0, faiss 1.6.3.
Can you provide me your environment settings? Thank you!

**Comment by yxgeee** [comment]
Hi,

My environment is: cuda 9.0, python 3.6.9, torch 1.3.1, faiss 1.6.3. I also tested with torch 1.1.0.
I haven't met your problems before. On which line did it get stuck?

**Comment by sxwenny** [comment]
I don't know which line it got stuck, but I think these two problems are related to sourece and target dataloader [https://github.com/pytorch/pytorch/issues/24927](url)

**Comment by yxgeee** [comment]
You can also try to use other `search_option` (default:0) for `compute_jaccard_distance()` to test if the problem is with the faiss. https://github.com/yxgeee/SpCL/blob/master/spcl/utils/faiss_rerank.py#L37

**Comment by sxwenny** [comment]
Okay, thank you!
